h/static/scripts/app-controller.js

```diff
@@ -46,37 +60,21 @@ module.exports = function AppController(
 
   // Reload the view when the user switches accounts
   $scope.$on(events.USER_CHANGED, function (event, data) {
+    $scope.auth = authStateFromUserID(data.userid);
+    $scope.accountDialog.visible = false;
+
     if (!data || !data.initialLoad) {
       $route.reload();
     }
   });
 
-  identity.watch({
-    onlogin: function (identity) {
-      // Hide the account dialog
-      $scope.accountDialog.visible = false;
-      // Update the current logged-in user information
-      var userid = auth.userid(identity);
-      var parsed = parseAccountID(userid);
-      angular.copy({
```

Comment:
This was previously using `angular.copy`, presumably to preserve the identify of the `$scope.auth` object. The new code doesn't do that -- are we sure that this behaviour doesn't need to be preserved?